src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/UpdateContactDetailsController.java

```diff
@@ -0,0 +1,110 @@
+
+public class UpdateContactDetailsController extends BaseController {
+
+    void persistOrgPolicies(Map<String, Object> caseData, CaseDetails originalDetails) {
+        caseData.put(APPLICANT_ORGANISATION_POLICY, originalDetails.getData().get(APPLICANT_ORGANISATION_POLICY));
+        caseData.put(RESPONDENT_ORGANISATION_POLICY, originalDetails.getData().get(RESPONDENT_ORGANISATION_POLICY));
+    }
+
+    boolean isIncludesRepresentationChange(Map<String, Object> caseData) {
+        return YES_VALUE.equals(caseData.get(INCLUDES_REPRESENTATION_CHANGE));
+    }
+
+
+    void handleApplicantRepresentationChange(Map<String, Object> caseData, boolean isContested) {
+        String nocParty = (String) caseData.get(NOC_PARTY);
+        if (APPLICANT.equalsIgnoreCase(nocParty)) {
+            removeApplicantSolicitorDetails(caseData, isContested);
+        }
+    }
+
+    void handleRespondentRepresentationChange(CaseDetails caseDetails, boolean isContested) {
+        Map<String, Object> caseData = caseDetails.getData();
+        String nocParty = (String) caseData.get(NOC_PARTY);
+        if (RESPONDENT.equalsIgnoreCase(nocParty)) {
+            removeRespondentDetails(caseData, caseDetails.getCaseTypeId());
+        }
+    }
+
+    void removeApplicantSolicitorDetails(Map<String, Object> caseData, boolean isContested) {
+        String applicantRepresented = nullToEmpty(caseData.get(APPLICANT_REPRESENTED));
+        if (applicantRepresented.equals(NO_VALUE)) {
+            caseData.remove(isContested ? CONTESTED_SOLICITOR_NAME : CONSENTED_SOLICITOR_NAME);
+            caseData.remove(isContested ? CONTESTED_SOLICITOR_FIRM : CONSENTED_SOLICITOR_FIRM);
+            caseData.remove(isContested ? CONTESTED_SOLICITOR_ADDRESS : CONSENTED_SOLICITOR_ADDRESS);
+            caseData.remove(isContested ? CONTESTED_SOLICITOR_PHONE : SOLICITOR_PHONE);
+            caseData.remove(isContested ? CONTESTED_SOLICITOR_EMAIL : SOLICITOR_EMAIL);
+            caseData.remove(isContested ? APP_SOLICITOR_AGREE_TO_RECEIVE_EMAILS_CONTESTED : APP_SOLICITOR_AGREE_TO_RECEIVE_EMAILS_CONSENTED);
+
+            caseData.remove(SOLICITOR_REFERENCE);
+            caseData.remove(isContested ? null : CONSENTED_SOLICITOR_DX_NUMBER);
+            caseData.remove(APPLICANT_ORGANISATION_POLICY);
+        }
+    }
+
+    void removeRespondentDetails(Map<String, Object> caseData, String caseTypeId) {
+        boolean isContested = caseTypeId.equalsIgnoreCase(CaseType.CONTESTED.getCcdType());
+        String respondentRepresented = isContested
+            ? (String) caseData.get(CONTESTED_RESPONDENT_REPRESENTED)
+            : (String) caseData.get(CONSENTED_RESPONDENT_REPRESENTED);
+        if (respondentRepresented.equals(YES_VALUE)) {
+            caseData.remove(RESPONDENT_ADDRESS);
+            caseData.remove(RESPONDENT_PHONE);
+            caseData.remove(RESPONDENT_EMAIL);
+            caseData.put(RESPONDENT_RESIDE_OUTSIDE_UK, NO_VALUE);
+        } else {
```

Comment:
One thing I need to confirm is the logic of `removeApplicantSolicitorDetails` and `removeRespondentDetails`. 

removeRespondentDetails seems to remove respondent information if they are represented. Do we ought to have the same for the applicant too?